multi.go

```diff
@@ -0,0 +1,137 @@
+package smt
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"hash"
+)
+
+var (
+	_ MultiStore = (*multi)(nil)
+	_ Store      = (*store)(nil)
+
+	// StoreCreator is a function that creates a new Store for the given MultiStore
+	StoreCreator = func(name string, ms MultiStore) (Store, MapStore) {
+		db := NewSimpleMap()
+		return NewStore(name, ms, db, sha256.New()), db
+	}
+)
+
+type multi struct {
```

Comment:
Should the fields be exported and have getters so you can plug and play your own and pass it as an interface into the store?